src/test/groovy/graphql/Issue1914.groovy

```diff
@@ -0,0 +1,81 @@
+package graphql
+
+import graphql.schema.DataFetcher
+import spock.lang.Specification
+
+class Issue1914 extends Specification {
+
+    def "default values in input objects are respected when variable is not provided"() {
+        given:
+        def spec = """type Query {
+            sayHello(arg: Arg! = {}): String
+        }
+        input Arg {
+            foo: String = "bar"
+        }
+        """
+        DataFetcher df = { dfe ->
+            Map arg = dfe.getArgument("arg")
+            return arg.get("foo")
+        } as DataFetcher
+        def graphQL = TestUtil.graphQL(spec, ["Query": ["sayHello": df]]).build()
+
+        when:
+        def result = graphQL.execute('{sayHello}')
+
+        then:
+        result.errors.isEmpty()
+        result.data == [sayHello: "bar"]
+
+    }
+
+    def "default values in input objects are overridden when variable is provided"() {
+        given:
+        def spec = """type Query {
+            sayHello(arg: Arg! = {foo: "brewery"}): String
+        }
+        input Arg {
+            foo: String = "bar"
+        }
+        """
+        DataFetcher df = { dfe ->
+            Map arg = dfe.getArgument("arg")
+            return arg.get("foo")
+        } as DataFetcher
+        def graphQL = TestUtil.graphQL(spec, ["Query": ["sayHello": df]]).build()
+
+        when:
+        def result = graphQL.execute('{sayHello}')
+
+        then:
+        result.errors.isEmpty()
+        result.data == [sayHello: "brewery"]
+
+    }
+
+    def "default values in input objects are overridden when variable is provided and otherwise are respected"() {
+        given:
+        def spec = """type Query {
+            sayHello(arg: Arg! = {field1: "F1ValOverride"}): String
+        }
+        input Arg {
+            field1: String = "F1V"
+            field2: String = "F2V"
+        }
+        """
+        DataFetcher df = { dfe ->
+            Map arg = dfe.getArgument("arg")
+            return arg.get("field1") + " & " + arg.get("field2")
+        } as DataFetcher
+        def graphQL = TestUtil.graphQL(spec, ["Query": ["sayHello": df]]).build()
+
+        when:
+        def result = graphQL.execute('{sayHello}')
+
+        then:
+        result.errors.isEmpty()
+        result.data == [sayHello: "F1ValOverride & F2V"]
```

Comment:
Can we get another test in place for when the inner field is named BUT the value is null

Reply:
do you something like 
"default values in input objects are overridden when null value is provided in input"?

or do you mean the default value provided for a field in input type is null?

Reply:
@bbakerman Added a test where one of the fields in input type have null value (below this test)